### DISCUSSION TITLE
Algolia search

### DISCUSSION THREAD

**Discussion by teamcrisis** [thuliteio/doks#120] [Ideas]
Any chance Algolia search will be added as an alternative or replacement?

It seems FlexSearch hasn't been actively developed in quite some time.

**Comment by h-enk** [comment]
I've been thinking about that too, but I'm not sure yet.

Algolia does offer a [free plan](https://www.algolia.com/pricing/) — with a limit of course. But, I do not want Dok's users to face costs when  crossing the limit. So, I'm not too keen to make it a default functionality. However, I'm considering writing instructions how to add Algolia as an extension /replacement of FlexSearch.

I'm aware that the latest commit to FlexSearch was about 15 months ago. I will contact @nextapps-de about the [0.7.0 release](https://github.com/nextapps-de/flexsearch/tree/0.7.0).

**Reply by teamcrisis** [reply]
Yes, having a guide to add Algolia would be great. I think their free plan would cover most small websites and for sites that exceed the free limits, the pricing is reasonable. 

I tried contacting the maintainer for FlexSearch via email last week, but I have not heard back.

**Comment by epogrebnyak** [comment]
Is Lunr any good as a search? https://github.com/olivernn/lunr.js/

вс, 7 февр. 2021 г., 18:09 Henk Verlinde <notifications@github.com>:

**Reply by h-enk** [reply]
Maybe, I'll have to look into it. 

See also [Benchmark Ranking](https://github.com/nextapps-de/flexsearch/tree/0.7.0#benchmark-ranking)

**Comment by teamcrisis** [comment]
I've also noticed that the size of the JS file is growing rapidly as I add sections to the index. For larger sites, this isn't ideal, so an external search service would be better if you have large documents or a large number of pages.

**Comment by h-enk** [comment]
I've been rethinking Doks search last night.

Long story short, I'm not happy with how FlexSearch is set up architecturally (won't go into details here, for now) — and more importantly how that works out for the user (won't go into details here either, for now). 

Re. Algolia, 

> I think their free plan would cover most small websites and for sites that exceed the free limits, the pricing is reasonable.

@teamcrisis, I think so too.

So, what it comes down to for Doks: I'm planning to replace FlexSearch w/ Algolia search.

And yes, Algolia sponsoring [getdoks.org](https://getdoks.org/) (and [gethyas.com](https://gethyas.com/)) also helps in this decision 😉

**Reply by h-enk** [reply]
See also: https://github.com/h-enk/doks/issues/130

**Reply by teamcrisis** [reply]
Glad to hear about the change.

You might want to take a look at [Forestry](https://forestry.io/docs/welcome/) as well. I like their search and overall documentation setup. 

They have a [repo](https://github.com/forestryio/forestry.io) as well.

**Comment by anvychow** [comment]
That's great to hear. Any update on adding a way to replace Flex with Algolia?

**Reply by h-enk** [reply]
Planned for [v0.3.0](https://github.com/h-enk/doks/milestone/2) — so in the works. I've been very busy lately, but expect to release [v0.2.0](https://github.com/h-enk/doks/milestone/1) soon. And, release v0.3.0 soon thereafter.

**Comment by almereyda** [comment]
There are also other practical search engines that can carry an external index, such as:

* https://github.com/meilisearch/MeiliSearch
* https://github.com/typesense/typesense
* https://github.com/quickwit-oss/tantivy

Eventually they are worth a look, too?

**Reply by h-enk** [reply]
Yeah, absolutely. Note that a guide on how to add [DocSearch](https://docsearch.algolia.com/) is in the making — like how it's implemented in [getdoks.org](https://getdoks.org/docs/)

**Reply by nk9** [reply]
That would be great to see.

**Reply by constantcopyandwork** [reply]
> Yeah, absolutely. Note that a guide on how to add [DocSearch](https://docsearch.algolia.com/) is in the making — like how it's implemented in [getdoks.org](https://getdoks.org/docs/)

Would we be able to apply that to blog post searches too?
Thanks!

**Reply by h-enk** [reply]
Yes, if it's a "technical blog" — see also [Who can apply?](https://docsearch.algolia.com/docs/who-can-apply)

**Reply by constantcopyandwork** [reply]
Oh, that's too bad... Know of any alternatives? Or would you recommend the Legacy version?
Thanks

**Reply by h-enk** [reply]
Yes, I would recommend to [run your own](https://docsearch.algolia.com/docs/legacy/run-your-own) then.
